core/server/master/src/main/java/alluxio/master/file/meta/SimpleInodeLockList.java

```diff
@@ -257,32 +278,22 @@ public LockMode getLockMode() {
 
   @Override
   public List<Inode> getLockedInodes() {
-    return mEntries.stream()
-        .filter(e -> e instanceof InodeEntry)
-        .map(e -> ((InodeEntry) e).getInode())
-        .collect(Collectors.toList());
+    return new ArrayList<>(mInodes);
   }
 
   @Override
   public Inode get(int index) {
-    int entryIndex = mEntries.get(0) instanceof InodeEntry ? index * 2 : index * 2 + 1;
-    return ((InodeEntry) mEntries.get(entryIndex)).getInode();
+    return mInodes.get(index);
```

Comment:
This will be slow with the linked list now?

I'd imagine most of the usage only wants the last index? Is that efficient for the implementation of the linked list?

[later] Maybe we should consider adding a `getLastInode` api.

Reply:
LinkedList is a double linked list, so getting the last index is O(1)

Reply:
looks like LinkedList.get() starts from the end that the index is closer to.